### PR TITLE
Full project

### PR DESCRIPTION
**Pull request by MickaelAlliel**

### CODE REVIEW

File: src/handlers/todo.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        return res.status(400).send({message: 'Missing required payload parameter <Title>'})
+    }
+    let todo = {};
+    todo.title = cleanString(req.body.title);
+    todo.completed = false;
+    const response = await global.Store.save(todo);
+    return res.status(200).send({message: response})
+};
+
+exports.UpdateTodo = async (req, res, next) => {
```
Comment (iglizer) [inline src/handlers/todo.js]: please use module.exports

File: src/utils/store.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+/**
+ * Will clear all completed todos
+ *
+ * @returns {Promise}
+ */
+Store.prototype.ClearCompleted = function() {
+    return new Promise((resolve, reject) => {
+        try {
+            const todos = todos.filter(todo => {
+                if (todo.completed) {
```
Comment (iglizer) [inline src/utils/store.js]: return !todo.completed

File: src/utils/store.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+/**
+ * Will toggle all todos
+ *
+ * @returns {Promise}
+ */
+Store.prototype.ToggleAll = function() {
+    return new Promise((resolve, reject) => {
+        try {
+            let todos = this.todos;
+            for (let i = 0; i < todos.length; i++) {
```
Comment (iglizer) [inline src/utils/store.js]: using of `for` loops is old imperative approach, better to use declarative style.  
https://codeburst.io/declarative-vs-imperative-programming-a8a7c93d9ad2

here we can use `todos.forEach`

File: src/static/js/requester.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        data: JSON.stringify({title: newData.title, completed: newData.completed}),
+        dataType: 'json'
+    })
+    .done(function() {
+        location.reload();
+    });
+}
+
+const updateTodo = (e) => {
+    if ($(e.target).hasClass('todo-label')) {
```
[the remark this answers is not in the document]
Comment (MickaelAlliel) [reply src/static/js/requester.js]: It is already split to functions -> getTodoData, sendUpdateRequest
I had it initially as 2 seperate completely, when toggling or when editing and Itiel told me it's better to combine the two. So what's the better way?

**Comment by MickaelAlliel** [comment]
I fixed everything, only thing that remain unchanged is this comment
If you want to have a look again, I tried to make it as best as I can understand from your tips :)
```
src/static/js/requester.js
+}
+
+const updateTodo = (e) => {
+    if ($(e.target).hasClass('todo-label')) {
  @iglizer
iglizer 2 hours ago  Member
it's too long, I would split it to few functions, event just to understand the flow here.

   @MickaelAlliel
MickaelAlliel 16 minutes ago  Member
It is already split to functions -> getTodoData, sendUpdateRequest
I had it initially as 2 seperate completely, when toggling or when editing and Itiel told me it's better to combine the two. So what's the better way?
```

File: src/static/js/requester.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (iglizer) [reply src/static/js/requester.js]: so split to "paragraphs" as we talked

**Pull request merged**